### DISCUSSION TITLE
Testing Threads Streamed Responses

### DISCUSSION THREAD

**Discussion by aprod-captiwate** [openai-php/client#442] [Q&A]
Hi!

I'm trying to test a streamed thread run with a given fixture:

```
    OpenAI::fake([
        CreateStreamedResponse::fake(fopen('file1.txt'), 'r')),
    ]);
```

But I got 
``` 
 Undefined array key "choices" at vendor/openai-php/client/src/Responses/Chat/CreateStreamedResponse.php:45
```

I figured it is because `CreateStreamedResponse` is for chat completions, not for threads, but I can't find an equivalent for Threads. Is it not yet implemented?

Thanks for the help!

**Comment by aprod-captiwate** [comment]
I think I figured what is missing: in the `ThreadRunStreamResponse.php` we need to use the `FakeableForStreamedResponse` trait, I made a workaround in my tests, until it will be fixed in the main lib. I will try to create a PR.

**Reply by Muhammad-Huzaifa7** [reply]
I am encountering an error "Unsupported operand types: string * int" when trying to use the OpenAI PHP Client to upload files and create assistants. I have run the commands again but the error still persists, can you please help me out.
